src/components/TokenDetailAction.js

```diff
@@ -0,0 +1,438 @@
+
+    return <TokenAction renderForm={renderForm} title={title} buttonName='Delegate' onPinSuccess={this.executeDelegate} {...this.props} />
+  }
+}
+
+
+class TokenDestroy extends React.Component {
+  /**
+   * destroyQuantity {number} Holds the quantity written by the user on the input (need to use state to fill the modal data)
+   */
+  state = {
+    destroyQuantity: 0,
+  }
+
+  /**
+   * Execute destroy method
+   *
+   * @param {string} pin PIN user wrote on modal
+   *
+   * @return {Object} Object with promise and success message
+   */
+  executeDestroy = (pin) => {
+    const label = this.props.action === 'destroy-mint' ? 'Mint' : 'Melt';
+    const array = this.props.authorityOutputs;
+    const data = [];
+    // Get the number of outputs the user requested to destroy in the expected format
+    for (let i=0; i<this.state.destroyQuantity; i++) {
+      data.push({
+        'tx_id': array[i].tx_id,
+        'index': array[i].index,
+        'address': array[i].decoded.address,
+        'token': this.props.token.uid
+      });
+    }
+    const promise = hathorLib.tokens.destroyAuthority(data, pin);
+    return { promise, message: `${label} outputs destroyed!`};
+  }
+
+  /**
+   * Called when clicking to destroy mint outputs.
+   * Validate if we have the quantity of outputs requested to destroy and open the PIN modal
+   *
+   * @return {string} Error message, in case of form invalid. Nothing, otherwise.
+   */
+  destroy = () => {
+    if (this.state.destroyQuantity > this.props.authorityOutputs.length) {
+      return `You only have ${this.props.authorityOutputs.length} mint ${hathorLib.helpers.plural(this.props.authorityOutputs.length, 'output', 'outputs')} to destroy.`;
+    }
+  }
+
+  render() {
+    const getDestroyBody = () => {
+      const quantity = parseInt(this.state.destroyQuantity, 10);
+      const type = this.props.action === 'destroy-mint' ? 'mint' : 'melt';
+      const plural = hathorLib.helpers.plural(quantity, 'output', 'outputs');
+
+      return (
+        <p>Are you sure you want to destroy <strong>{quantity} {type}</strong> authority {plural}?</p>
+      )
+    }
+
+    const renderForm = () => {
+      return (
+        <div className="row">
+          <div className="form-group col-6">
+            <label>How many {this.props.action === 'destroy-mint' ? 'mint' : 'melt'} outputs you want to destroy?</label>
+            <input required type="number" className="form-control" min="1" step="1" ref={this.destroyQuantity} onChange={(e) => this.setState({ destroyQuantity: e.target.value })} />
+          </div>
+        </div>
+      );
+    }
+
+    const title = `Destroy ${this.props.action === 'destroy-mint' ? 'Mint' : 'Melt'}`;
+
+    return <TokenAction renderForm={renderForm} validateForm={this.destroy} title={title} buttonName='Destroy' onPinSuccess={this.executeDestroy} pinBodyTop={getDestroyBody()} {...this.props} />
+  }
+}
+
+class TokenAction extends React.Component {
```

Comment:
Add description of what the component does. In this case, I haven't gotten what it does, lol. :)

Reply:
Done